jujucrashdump/crashdump.py

```diff
@@ -180,22 +179,51 @@ def run_addons(self):
 
     def run_listening(self):
         pull_location = "/tmp/{uniq}/cmd_output".format(uniq=self.uniq)
-        run_cmd('juju run --all "mkdir -p %s"' % pull_location)
-        listening_cmd = """sudo netstat -taupn | grep LISTEN 2>/dev/null > \
-{pull_location}/listening.txt || true""".format(pull_location=pull_location)
-        run_cmd("""timeout %ds juju run --all 'sh -c "%s"'""" % (
-            self.timeout, listening_cmd))
+        self._run_all('mkdir -p {pull_location};'
+                      'sudo netstat -taupn | grep LISTEN 2>/dev/null'
+                      ' > {pull_location}/listening.txt || true'
+                      ''.format(pull_location=pull_location))
+
+    def run_journalctl(self):
+        for service in self.journalctl or []:
+            logdir = '/tmp/{uniq}/journalctl'.format(uniq=self.uniq)
+            logfile = '{logdir}/{service}.log'.format(logdir=logdir,
+                                                      service=service)
+            self._run_all('mkdir -p {logdir};'
+                          'journalctl -u {service} > {logfile};'
+                          '[ $(wc -c < {logfile}) = 17 ] && rm {logfile};'
```

Comment:
TIL that `-- No entries --` is 17 bytes long, but why do this instead of a pattern match?

Reply:
Seems expensive to dump the whole file to disk just to see if it is 17 bytes long. Is there not a return value or something we can use here?

Reply:
Unfortunately, no.  journalctl is kinda dumb and just exits 0 even if you feed it a completely invalid service name.